dis_snek/models/discord_objects/scheduled_event.py

```diff
@@ -0,0 +1,204 @@
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+import attr
+from attr.converters import optional
+from dis_snek.const import MISSING
+from dis_snek.models.discord import DiscordObject
+
+from dis_snek.models.snowflake import Snowflake_Type, to_snowflake
+from dis_snek.models.timestamp import Timestamp
+from dis_snek.utils.attr_utils import define
+from dis_snek.utils.converters import timestamp_converter
+
+
+from enum import IntEnum
+
+if TYPE_CHECKING:
+    from dis_snek.client import Snake
+    from dis_snek.models.discord_objects.channel import GuildStageVoice, GuildVoice
+    from dis_snek.models.discord_objects.guild import Guild
+    from dis_snek.models.discord_objects.user import Member
+    from dis_snek.models.discord_objects.channel import TYPE_ALL_CHANNEL
+    from dis_snek.models.discord_objects.user import User
+    from dis_snek.models.snowflake import Snowflake_Type
+
+
+class ScheduledEventPrivacyLevel(IntEnum):
+    """The privacy level of the scheduled event."""
+
+    # Discord only has one at the momment.
+    GUILD_ONLY = 2
+
+
+class ScheduledEventEntityType(IntEnum):
+    """The type of entity that the scheduled event is attached to."""
+
+    STAGE_INSTANCE = 1
+    """ Stage Channel """
+    VOICE = 2
+    """ Voice Channel """
+    EXTERNAL = 3
+    """ External URL """
+
+
+class ScheduledEventStatus(IntEnum):
+    """The status of the scheduled event."""
+
+    # Discord only has one at the momment.
+    SCHEDULED = 1
+    ACTIVE = 2
+    COMPLETED = 3
+    CANCELED = 4
+
+
+@define()
+class ScheduledEventUser(DiscordObject):
```

Comment:
Why is this object necessary? Afiak there is no extra information over a standard `member` object when a member is part of an event?

Reply:
Could you help me out? I am not sure what else I would do. I did this because I just saw how the team.py file did it.

Reply:
Teams.py uses this paradigm because it adds additional information to the objects. Here you're adding nothing so it's a redundant object

Easiest solution would be to store the list of participant snowflakes, then make a property which returns the list of members using a generator

Reply:
Tho, Events could easily have thousands of participants... Maybe an async iterator?

Reply:
Sorry, I don't think I am comfortable adding something like this yet. I am gonna have to pass this on to someone else and watch what they do so I can see how this would be done

Reply:
From what I'm understanding, you want something like this?
( this is pseudocode, me no dis-snek :( )

```py
    ...

    @property
    async def users(self):
        _users: list = [models.user.User(user) for user in self.participants]
```